jeansaad/chalet
# Ability to edit server environment variables

## Context

Add the ability to fully edit the environment variables of a server within the user interface.

## Screenshots

### Content Menu

<img width="257" alt="image" src="https://user-images.githubusercontent.com/9013008/191595997-4ede6d7e-9db7-4999-9bac-0ca630eaa4a8.png">

### Settings Component

<img width="1415" alt="image" src="https://user-images.githubusercontent.com/9013008/191596206-e2866488-2616-43d3-af96-651017c460fd.png">

## Test Plan




1. Start your chalet process.
2. Go to the user interface
3. Select a server (whether on, or off)
4. Click the cog in the upper righthand corner.
5. Edit config however you want
6. Click save and ensure the new config file `~/.chalet/services/YOUR_SERVER.json` is as expected.
7. Attempt to save with improper json and make sure this is gracefully handled.
8. Check empty state (config file has no `env` property).
9. Check the cancel button simply closes the settings menu, and makes no file changes.
## Items requiring extra attention
